cli/azd/pkg/tools/dotnet/dotnet.go

```diff
@@ -178,18 +187,45 @@ func (cli *dotNetCli) PublishContainer(
 		"/t:PublishContainer",
 		fmt.Sprintf("-p:ContainerImageName=%s", imageName),
 		fmt.Sprintf("-p:ContainerRegistry=%s", server),
+		"--getProperty:GeneratedContainerConfiguration",
 	)
 
 	runArgs = runArgs.WithEnv([]string{
 		fmt.Sprintf("SDK_CONTAINER_REGISTRY_UNAME=%s", username),
 		fmt.Sprintf("SDK_CONTAINER_REGISTRY_PWORD=%s", password),
 	})
 
-	_, err := cli.commandRunner.Run(ctx, runArgs)
+	result, err := cli.commandRunner.Run(ctx, runArgs)
 	if err != nil {
-		return fmt.Errorf("dotnet publish on project '%s' failed: %w", project, err)
+		return 0, fmt.Errorf("dotnet publish on project '%s' failed: %w", project, err)
 	}
-	return nil
+	port, err := cli.getTargetPort(result.Stdout)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get target port on project '%s': %w", project, err)
+	}
+
+	return port, nil
+}
+
+func (cli *dotNetCli) getTargetPort(
+	output string,
+) (int, error) {
+	var configOutput responseContainerConfiguration
+	err := json.Unmarshal([]byte(output), &configOutput)
+	if err != nil {
+		return 0, fmt.Errorf("unmarshal configuration output: %w", err)
+	}
+	var exposedPortOutput string
+	for key, _ := range configOutput.Config.ExposedPorts {
+		exposedPortOutput = key
+	}
+	exposedPortRegexp := regexp.MustCompile(`\d+`)
+	exposedPortString := exposedPortRegexp.FindString(exposedPortOutput)
+	exposedPort, err := strconv.Atoi(exposedPortString)
+	if err != nil {
+		return 0, fmt.Errorf("failed to convert target port '%s' to integer: %w", exposedPortString, err)
+	}
+	return exposedPort, nil
```

Comment:
I would change `exposedPort` to be string, so you don't need to call `Atoi`.

And, `Config.ExposedPorts` is a `map[string]`, which means there might be more than one port in there.  So, change the return type to be `[]string`.  Whoever calls that `getTargetPort()` should take the decision of either taking just the first element or looking at all.

Now, each `key` is built as `{port}/{protocol}`.  Both, the port and the protocol are important for whoever that calls this method.  So, let's make `getTargerPort()` to return `[]targetPort` and define `targerPort` as:

```go
type targetPort struct {
     port string
     protocol string
}
```

Then, split each key with `/` char and create a targerPort and put it in the list.